internal/cmd/beta/security-group-rule/create/create_test.go

```diff
@@ -0,0 +1,353 @@
+	request = request.CreateSecurityGroupRulePayload(iaas.CreateSecurityGroupRulePayload{
+		Direction: utils.Ptr("ingress"),
+	})
+	for _, mod := range mods {
+		mod(&request)
+	}
+	return request
+}
+
+func fixturePayload(mods ...func(payload *iaas.CreateSecurityGroupRulePayload)) iaas.CreateSecurityGroupRulePayload {
+	payload := iaas.CreateSecurityGroupRulePayload{
+		Direction:   utils.Ptr("ingress"),
+		Description: utils.Ptr("example-description"),
+		Ethertype:   utils.Ptr("ether"),
+		IcmpParameters: &iaas.ICMPParameters{
+			Code: utils.Ptr(int64(0)),
+			Type: utils.Ptr(int64(8)),
+		},
+		IpRange: utils.Ptr("10.1.2.3"),
+		PortRange: &iaas.PortRange{
+			Max: utils.Ptr(int64(24)),
+			Min: utils.Ptr(int64(22)),
+		},
+		Protocol: &iaas.CreateProtocol{
+			Int64:  utils.Ptr(int64(1)),
+			String: utils.Ptr("icmp"),
+		},
+		RemoteSecurityGroupId: utils.Ptr(testRemoteSecurityGroupId),
+	}
+	for _, mod := range mods {
+		mod(&payload)
+	}
+	return payload
+}
+
+func TestParseInput(t *testing.T) {
+	tests := []struct {
+		description   string
+		flagValues    map[string]string
+		isValid       bool
+		expectedModel *inputModel
+	}{
+		{
+			description: "base",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				delete(flagValues, portRangeMaxFlag)
+				delete(flagValues, portRangeMinFlag)
+				delete(flagValues, protocolNumberFlag)
+			}),
+			isValid: true,
+			expectedModel: fixtureInputModel(func(model *inputModel) {
+				model.PortRangeMax = nil
+				model.PortRangeMin = nil
+				model.ProtocolNumber = nil
+			}),
+		},
+		{
+			description: "required only",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				delete(flagValues, descriptionFlag)
+				delete(flagValues, etherTypeFlag)
+				delete(flagValues, icmpParameterCodeFlag)
+				delete(flagValues, icmpParameterTypeFlag)
+				delete(flagValues, ipRangeFlag)
+				delete(flagValues, portRangeMaxFlag)
+				delete(flagValues, portRangeMinFlag)
+				delete(flagValues, remoteSecurityGroupIdFlag)
+				delete(flagValues, protocolNumberFlag)
+				delete(flagValues, protocolNameFlag)
+			}),
+			isValid: true,
+			expectedModel: fixtureInputModel(func(model *inputModel) {
+				model.Description = nil
+				model.EtherType = nil
+				model.IcmpParameterCode = nil
+				model.IcmpParameterType = nil
+				model.IpRange = nil
+				model.PortRangeMax = nil
+				model.PortRangeMin = nil
+				model.RemoteSecurityGroupId = nil
+				model.ProtocolNumber = nil
+				model.ProtocolName = nil
+			}),
+		},
+		{
+			description: "direction missing",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				delete(flagValues, directionFlag)
+			}),
+			isValid: false,
+		},
+		{
+			description: "protocol is icmp and parameters are missing",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				delete(flagValues, icmpParameterCodeFlag)
+				delete(flagValues, icmpParameterTypeFlag)
+			}),
+			isValid: false,
```

Comment:
Same here. That is what I mean, in order to check if the test is working correctly you have to set "isValid" manually to "true" and need to check if the raised error matches the test. This is cumbersome in my opinion and error prone.